jobserver/api/jobs.py

```diff
@@ -119,6 +119,16 @@ def post(self, request, *args, **kwargs):
                 else:
                     updated_job_ids.append(str(job.id))
 
+                if "internal error" in job.status_message.lower():
```

Comment:
Am I right that this string is generated in job-runner [here](https://github.com/opensafely-core/job-runner/search?q=%22internal+error%22+filename%3Arun.py)?

This seems like a good immediate solution to a current problem, but I guess that ideally we'd want something a bit more structured so we're not depending on string matching?

Reply:
I've just seen Dave's comment. :+1:

Reply:
@benbc We already have the basic building blocks to do this, but just aren't actually using them. Jobs have a `status_code` field whose values are defined [here](https://github.com/opensafely-core/job-runner/blob/4ef28a9c2e64d523439536c21663447c3f28bd86/jobrunner/models.py#L26-L36) and which is pushed up to job-server along with everything else. We need to add an `internal_error` code and then set that in the appropriate place.

Reply:
Great.